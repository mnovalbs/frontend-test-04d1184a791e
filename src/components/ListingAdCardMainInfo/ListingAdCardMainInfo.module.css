.mainInfo {
  display: flex;
  gap: 16px;
  align-items: center;
}

.title {
  font-weight: 600;
  font-size: 23px;
  line-height: 31px;
  margin: 0;
  color: #1a2258;
}

.address {
  font-weight: 400;
  font-size: 14px;
  line-height: 19px;
}
