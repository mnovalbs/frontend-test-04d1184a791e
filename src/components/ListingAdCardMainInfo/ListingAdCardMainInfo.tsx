import { Ad } from "types/Ad";

import styles from "./ListingAdCardMainInfo.module.css";

function ListingAdCardMainInfo({
  title,
  address
}: Pick<Ad, "title" | "address">) {
  return (
    <div className={styles.mainInfo}>
      <img
        src="/icons/building-icon.svg"
        alt="building"
        width={40}
        height={40}
      />
      <div>
        <h1 className={styles.title}>{title}</h1>
        <div className={styles.address}>{address}</div>
      </div>
    </div>
  );
}

export default ListingAdCardMainInfo;
