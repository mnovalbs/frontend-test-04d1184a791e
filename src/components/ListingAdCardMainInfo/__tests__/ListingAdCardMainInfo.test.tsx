import { render } from "@testing-library/react";
import mockAd from "contants/mockAd";
import ListingAdCardMainInfo from "../ListingAdCardMainInfo"

describe("ListingAdCardMainInfo", () => {
  it('renders main info', () => {
    const { container } = render(<ListingAdCardMainInfo address={mockAd.address} title={mockAd.title} />)

    expect(container).toMatchSnapshot()
  })
});