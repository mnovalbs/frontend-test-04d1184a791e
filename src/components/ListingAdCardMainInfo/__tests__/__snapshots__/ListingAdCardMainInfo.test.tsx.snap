// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListingAdCardMainInfo renders main info 1`] = `
<div>
  <div
    class="mainInfo"
  >
    <img
      alt="building"
      height="40"
      src="/icons/building-icon.svg"
      width="40"
    />
    <div>
      <h1
        class="title"
      >
        Forett At Bukit Timah
      </h1>
      <div
        class="address"
      >
        70 Anchorvale Crescent Â· D21
      </div>
    </div>
  </div>
</div>
`;
